### PR TITLE
[IMP] account: add column header name in account portal invoices template

### PR DESCRIPTION
**Pull request by CLaurelB**
In line with the approach outlined in [1], where element accessibility was improved by providing explicit names for easier selection and inheritance, this commit applies the same logic to the account portal invoices. By adding column header names, we ensure that these elements can be safely referenced without relying on positional selectors, thus avoiding potential issues when the element structure changes.

[1] https://github.com/odoo/enterprise/pull/63957

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by desdelinux** [comment]
Hi @JulienAlardot 

Could your review, please

**Review by desdelinux — COMMENTED** [review]
LGTM

**Review by luisg123v — COMMENTED** [review]
LGTM :+1:

**Review by Megaaaaaa — COMMENTED** [review]
Hello :wave:

I want to say that the diff LGTM and it is ready for merge but the issue is that it is useless diff at the moment. You don't use this improved version of the template so this commit is not useful. Maybe if you could include changes that needed this (or link a PR that uses it), it would justify the diff and i could approve it but right now, except for if i missed something, I don't think we need it. :thinking: Don't hesitate to tell me if I'm wrong tho :smile:

**Comment by CLaurelB** [comment]
> Hello 👋
> 
> I want to say that the diff LGTM and it is ready for merge but the issue is that it is useless diff at the moment. You don't use this improved version of the template so this commit is not useful. Maybe if you could include changes that needed this (or link a PR that uses it), it would justify the diff and i could approve it but right now, except for if i missed something, I don't think we need it. 🤔 Don't hesitate to tell me if I'm wrong tho 😄

Hello @Megaaaaaa 

I added a commit to update an XPath related to the suggested change. Would it be enough to fulfill your requirement?

Best regards,
cc. @luisg123v @desdelinux

**Review by Megaaaaaa — APPROVED** [review]
Yes thank you, the improvement looks justified now, way better imo.

Thank you! I'll send it for r+ :rocket:

**Review by oco-odoo — CHANGES_REQUESTED** [review]
Wouldn't it be safer to do that in master only ?

File: addons/account_payment/views/account_portal_templates.xml
Comment (CLaurelB) [reply addons/account_payment/views/account_portal_templates.xml]: I left the commit adding the names in this PR and moved the other commit to a PR for master: https://github.com/odoo/odoo/pull/183860

Comment (oco-odoo) [reply addons/account_payment/views/account_portal_templates.xml]: @CLaurelB Rather move both to master (on a branch with the same name, so that runbot synchronizes it with the enterprise counterpart). Merging this in 17.0 already has no real value ;  and is error-prone. Imagine some bugfix or new module introduced in 17.0 wants to override the view. The dev will see the names and use them. But actually, those names won't be set in every db, since they require a module update => this will break.

Comment (luisg123v) [reply addons/account_payment/views/account_portal_templates.xml]: Hi @oco-odoo,

> Merging this in 17.0 already has no real value

I consider this actually adds value. The problem is, these elements can't be reached on a deterministic manner, which is why we need this change. This is an usually accepted change in stable, a few examples would be:
- https://github.com/odoo/odoo/pull/59453
- https://github.com/odoo/odoo/pull/37769

> so that runbot synchronizes it with the enterprise counterpart

In this case, there's no enterprise PR involved.

> Imagine some bugfix or new module introduced in 17.0 wants to override the view. The dev will see the names and use them. But actually, those names won't be set in every db, since they require a module update

I think, in such a case, the dev won't be able to reach elements using deterministic xpaths anyway. Which, depending on the PoV, could be even worse because the new module could inadvertently produce incorrect or undesired behaviors.

What do you think?

Regards,

**Review by oco-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**
